### DISCUSSION TITLE
Shape of Problem.jacobian(free) // shape of Problem.jacobianstructure()

### DISCUSSION THREAD

**Discussion by Peter230655** [csu-hmc/opty#280] [Q&A]
1.
In the opty documentation release 1.4.0.dev0 on page 129 it says:
jacobian(free)
Returns.....
Return type
ndarray.shape((2n + q + r + s) * (n(N-1) + o ), )
but in my tests it returns
ndarray.shape((2n + q + r + s) * (n(N-1)) + o, ) The bracket to the right of o is moved to the left.

**Is there a mistale in my tests or is this a typo in the documentation?**

2.
When I tested Problem.jacobianstructure()
also the shape did not seem to match the shape given on page 129, but I found no alternative formula.

**Comment by moorepants** [comment]
Probably a typo.

**Reply by moorepants** [reply]
Yes it is a type. `o` is the number of instance constraints which is just added to the other EoM based constraints.

**Comment by Peter230655** [comment]
Thanks!
The meaning of `o` is explained in the docs.
Should I close this and make it an issue, so you remember when you write the next update on the documentation?
Or, is it possible that I make a PR on the documentation?

**Reply by moorepants** [reply]
>  Or, is it possible that I make a PR on the documentation?

That would be most ideal.

**Reply by Peter230655** [reply]
.....but you must give me a hint how to get started on doing this. :-)
What I mean is this: I now know, I think, how to push examples. I know how to change something in direct_collocation.py
But I have no idea how to propose a change in the documentation.

I looked wherever I could think of to find the document where this typo might be, but no luck :-(

**Comment by Peter230655** [comment]
This was corrected in the docstrings, I will close it.
